core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -454,14 +455,21 @@ private[spark] object ResourceUtils extends Logging {
     if (limitingResource.nonEmpty && !limitingResource.equals(ResourceProfile.CPUS)) {
       if ((taskCpus * maxTaskPerExec) < cores) {
         val resourceNumSlots = Math.floor(cores/taskCpus).toInt
-        val message = s"The configuration of cores (exec = ${cores} " +
-          s"task = ${taskCpus}, runnable tasks = ${resourceNumSlots}) will " +
-          s"result in wasted resources due to resource ${limitingResource} limiting the " +
-          s"number of runnable tasks per executor to: ${maxTaskPerExec}. Please adjust " +
-          "your configuration."
         if (sparkConf.get(RESOURCES_WARNING_TESTING)) {
+          val message = s"The configuration of cores (exec = ${cores} " +
```

Comment:
Same here. We can have only one message and use `message.message` in throwing exception.